src/groupeddataframe/groupeddataframe.jl

```diff
@@ -581,7 +611,8 @@ Base.haskey(gd::GroupedDataFrame, key::Union{Signed,Unsigned}) =
 Get a group based on the values of the grouping columns.
 
 `key` may be a `NamedTuple` or `Tuple` of grouping column values (in the same
-order as the `cols` argument to `groupby`).
+order as the `cols` argument to `groupby`). It may also be an `AbstractDict`, in which case the 
```

Comment:
also please add it can be `GroupKey` as in the implementation you allow it (and I think it is OK to allow it - just please make sure we test also this case)